lm_tournament_eval/api/tournament.py

```diff
@@ -201,56 +209,73 @@ def run_tournament(self):
+                                                  self.verbosity,
+                                                  cmd_filter=filter,
+                                                  gen_kwargs=self.config.gen_kwargs)
+
+            self.scheduler.set_task_size(len(task_dict[task_name].eval_docs))
+
+            original_task = deepcopy(task_dict[task_name])
+
+            for match_schedule in self.scheduler:
+                logging.info(f"Current match indices: {match_schedule}")
+
+                subtask = create_subtask(original_task, schedule=match_schedule)
+                
+
+                requests0, padding_reqests0 = create_requests(model0, 
+                                                              task=subtask,
+                                                              limit=self.config.limit)
+
+                results0 = self.tournament_evaluate(model=self.config.model0_name,
+                                                    lm=model0,
+                                                    model_args=self.config.model0_args,
+                                                    requests=requests0,
+                                                    task=subtask,
+                                                    eval_tasks=eval_tasks,
+                                                    task_dict=task_dict,
+                                                    padding_requests=padding_reqests0,
+                                                    batch_size=self.config.batch_size,
+                                                    device=self.config.device,
+                                                    limit=self.config.limit
+                                                )
+                        
+                requests1, padding_reqests1 = create_requests(model1,
+                                                              task=subtask,
+                                                              limit=self.config.limit)
+
+                results1 = self.tournament_evaluate(model=self.config.model1_name,
+                                                    lm=model1,
+                                                    model_args=self.config.model1_args,
+                                                    requests=requests1,
+                                                    task=subtask,
+                                                    eval_tasks=eval_tasks,
+                                                    task_dict=task_dict,
+                                                    padding_requests=padding_reqests1,
+                                                    batch_size=self.config.batch_size,
+                                                    device=self.config.device,
+                                                    limit=self.config.limit
+                                                )
+
+                rounds_per_task = []
+                match_results = {}
```

Comment:
we're not using `match_results` anywhere but shouldn't it's creation get moved up? or else it is reset every time between tasks?